components/Collections/src/MDCCollectionViewFlowLayout.m

```diff
@@ -391,13 +391,28 @@ - (MDCCollectionViewLayoutAttributes *)updateSupplementaryViewAttribute:
   // view rows.
   CGRect insetFrame = attr.frame;
   if (!CGRectIsEmpty(insetFrame)) {
-    UIEdgeInsets insets = [self insetsAtSectionIndex:attr.indexPath.section];
+    UIEdgeInsets insets = self.sectionInset;
```

Comment:
Good catch, I can switch the implementation to start with the internals and then call the delegate.